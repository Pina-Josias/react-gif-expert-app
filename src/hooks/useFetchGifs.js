import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

const useFetchGifs = (query) => {
  const [state, setstate] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(query).then((gifs) => {
      setstate({
        data: gifs,
        loading: false,
      });
    });
  }, [query]);

  return state;
};

export { useFetchGifs };
