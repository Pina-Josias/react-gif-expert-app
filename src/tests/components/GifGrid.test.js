import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas sobre el componente <GifGrid/>", () => {
  test("debe mostrar el componente <GifGrid/> correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={"Saitama"} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar items cuando se carga la data", () => {
    const gifs = [
      {
        id: "34234",
        url: "http://localhost/name",
        title: "test",
      },
      {
        id: "1234",
        url: "http://localhost/name",
        title: "test",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={"Saitama"} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
