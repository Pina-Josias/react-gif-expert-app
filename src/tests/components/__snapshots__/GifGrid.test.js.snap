// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid/> debe mostrar el componente <GifGrid/> correctamente 1`] = `
<Fragment>
  <h3>
    Saitama
  </h3>
  <div
    className="card-grid animate__animated animate__zoomInUp"
  >
    Cargando...
  </div>
</Fragment>
`;

exports[`Pruebas sobre el componente <GifGrid/> debe mostrar items cuando se carga la data 1`] = `
<Fragment>
  <h3>
    Saitama
  </h3>
  <div
    className="card-grid animate__animated animate__zoomInUp"
  >
    <GifGridItem
      id="34234"
      key="34234"
      title="test"
      url="http://localhost/name"
    />
    <GifGridItem
      id="1234"
      key="1234"
      title="test"
      url="http://localhost/name"
    />
  </div>
</Fragment>
`;
